tools/recording/record_common.py

```diff
@@ -144,7 +145,11 @@ def measure_configurations(self, args, num_configs):
                     # only for conv2d and linear replace the features with the ones obtained from main_generator
                     sample = params_generator.generate_sample()
                     config = list(self._index_to_config(args, config_id))
-                    config[len(config) - len(sample) :] = sample
+                    # for bachnorm, config is (batch, channel, image_size) and we need to replace channel := pos[1]
+                    if self._op_name == "batch_norm":
+                        config[1] = sample[0]
```

Comment:
Why is this logic implemented here but not `record_batchnorm.py`? It seems like we just need to modify the `index_to_config` function to avoid adding a special case here.

Reply:
sry, I changed the order, now it should be ok